frontend/src/app/ReusableComponents/UserAuth.tsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react'
+import { dm_Serif_Display_italic } from '../Fonts/DM_Serif_Display'
+import GoogleSignupOption from '../Sign-up/GoogleSignupOption';
+import { Roboto_Font } from '../Fonts/Roboto';
+interface IUserAuthProps {
+  username: string;
+  password: string;
+  authType:string;
+  setUsername: (value:string) => void;
+  setPassword: (value:string) => void;
+  handleAuth: (e:React.FormEvent) => void
+}
+
+export default function UserAuth({username,password,authType,setUsername, setPassword, handleAuth}:IUserAuthProps) {
```

Comment:
nit: i feel like destructuring all of the prop values up in the function args is kinda messy and long. it may be better to just destructure within the function?
```
export default function UserAuth(props: IUserAuthProps) { ... }
```
```
const {
    username,
    password,
    authType,
    setUsername,
    setPassword,
    handleAuth
``` = props;